sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -206,6 +210,15 @@ trait CodegenSupport extends SparkPlan {
   def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     throw new UnsupportedOperationException
   }
+
+  /* for optimization */
+  var shouldStopRequired: Boolean = false
+
+  def isShouldStopRequired: Boolean = {
+    if (shouldStopRequired) return true
```

Comment:
Can you just write `shouldStopRequired || (this.parent != null && this.parent.isShouldStopRequired)`?